src/block.cpp

```diff
@@ -1,18 +1,20 @@
 #include "block.h"
 
 Block::Block()
+: id(0)
+, cells()
+, cellSize(30)
+, rotationState(0)
+, rowOffset(0)
+, columnOffset(0)
+{}
+
+void Block::draw(int offsetX, int offsetY) const
 {
-    cellSize = 30;
-    rotationState = 0;
-    colors = getCellColors();
-    rowOffset = 0;
-    columnOffset = 0;
-}
-
-void Block::draw(int offsetX, int offsetY)
-{
-    std::vector<Position> tiles = getCellPositions();
-    for (Position item : tiles)
+    const auto& colors(getCellColors());
```

Comment:
getCellColor will now return **CONST REFERENCE** to a (cached) vector for all Cell Colors. So you may get rid off all local color storages which are spread around other classes! Think in responsibilities! ONE Function / method -> ONE responsibility! Gain access a to an **UNIQUE** instance (here a singleton) of all colours that is part of `getAllColors()`